File: src/web/types/Intent.ts
/**
 * The intent of calling the REST API.
 */
export enum Intent {

  /**
   * To get the full detail of the entities.
   */
  FULL_DETAIL = 'FULL_DETAIL',

  /**
   * To submit the feedback session as instructors.
   */
  INSTRUCTOR_SUBMISSION = 'INSTRUCTOR_SUBMISSION',

  /**
   * To submit the feedback session as students.
   */
  STUDENT_SUBMISSION = 'STUDENT_SUBMISSION',

  /**
   * To view the feedback session results as instructors.
   */
  INSTRUCTOR_RESULT = 'INSTRUCTOR_RESULT',

  /**
   * To view the feedback session results as students.
   */
  STUDENT_RESULT = 'STUDENT_RESULT',

}

File: src/main/java/teammates/ui/request/Intent.java
package teammates.ui.request;

/**
 * The intent of calling the REST API.
 */
public enum Intent {

    /**
     * To get the full detail of the entities.
     */
    FULL_DETAIL,

    /**
     * To submit the feedback session as instructors.
     */
    INSTRUCTOR_SUBMISSION,

    /**
     * To submit the feedback session as students.
     */
    STUDENT_SUBMISSION,

    /**
     * To view the feedback session result as instructors.
     */
    INSTRUCTOR_RESULT,

    /**
     * To view the feedback session result as students.
     */
    STUDENT_RESULT,

}
